### PR TITLE
feature add check connection

### PR DESCRIPTION
**Pull request by HyperionsThs**

### CODE REVIEW

File: app/src/main/java/com/hyperion/ths/marvel_03/ui/main/GetPointListener.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.hyperion.ths.marvel_03.ui.main;
+
+/**
+ * Created by ths on 12/06/2017.
+ */
+
+public interface GetPointListener {
```
Comment (daolq3012) [inline app/src/main/java/com/hyperion/ths/marvel_03/ui/main/GetPointListener.java]: Please rename of this interface map with function It do
u can use
``NetworkChangeListener``

File: app/src/main/java/com/hyperion/ths/marvel_03/ui/main/MainActivity.java [hunk 2]
```diff
@@ -12,15 +14,29 @@
 
 public class MainActivity extends AppCompatActivity {
     private MainViewModel mMainViewModel;
+    private MainReceiver mMainReceiver = new MainReceiver();
```
Comment (daolq3012) [inline app/src/main/java/com/hyperion/ths/marvel_03/ui/main/MainActivity.java]: Pls move initialization of this variable to ``onCreate`` method

File: app/src/main/java/com/hyperion/ths/marvel_03/ui/main/MainActivity.java [hunk 3]
```diff
@@ -12,15 +14,29 @@
...
+        mMainViewModel = new MainViewModel(mainViewPageAdapter, mMainReceiver);
 
         binding.setViewModel(mMainViewModel);
     }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        MainActivity.this.registerReceiver(mMainReceiver,
+                new IntentFilter("android.net.conn.CONNECTIVITY_CHANGE"));
```
Comment (daolq3012) [inline app/src/main/java/com/hyperion/ths/marvel_03/ui/main/MainActivity.java]: pls define ``"android.net.conn.CONNECTIVITY_CHANGE"`` as a constant

File: app/src/main/java/com/hyperion/ths/marvel_03/ui/main/GetPointListener.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    void setOnGetPointReceiver(OnGetPointReceiver onGetPointReceiver);
+
+    /**
+     * Get point listener
+     */
+    interface OnGetPointReceiver {
+        void getPoint(int point);
```
Comment (daolq3012) [inline app/src/main/java/com/hyperion/ths/marvel_03/ui/main/GetPointListener.java]: change ``void getPoint(int point);`` -> ``void onNetworkChange(boolean isConnected);``

File: app/src/main/java/com/hyperion/ths/marvel_03/utils/Constant.java [hunk 5]
```diff
@@ -23,6 +23,7 @@ private Constant() {
     public static final int CONNECTION_TIMEOUT = 60;
     public static final String BUNDLE_HERO = "BUNDLE_HERO";
     public static final int POINT = 0;
+    public static final String CONNECTIVITY_CHANGE = "android.net.conn.CONNECTIVITY_CHANGE";
```
Comment (DoanVanToan) [inline app/src/main/java/com/hyperion/ths/marvel_03/utils/Constant.java]: pls use CONNECTIVITY_ACTION in android.net.ConnectivityManager

**Pull request merged**